esrally/rally.py

```diff
@@ -587,8 +587,12 @@ def runtime_jdk(v):
 
 def derive_sub_command(args):
     sub_command = args.subcommand
-    # TODO: Can we select a subcommand by default?
-    return sub_command if sub_command is not None else "race"
+    if sub_command is None:
+        logger = logging.getLogger(__name__)
+        console.warn("Invoking Rally without a subcommand is deprecated. Specify the 'race' subcommand explicitly.", logger=logger)
```

Comment:
I wonder, have we ever considered https://docs.python.org/3/library/warnings.html for such warnings?

Reply:
Should we specify that the subcommand `race` will be required starting with Rally 2.1.0?

Reply:
I attemped to use the warnings module now but there are two issues:

1. We configure Rally to redirect warnings to the log. We do this because of urllib3 which would otherwise write to the terminal (see https://github.com/elastic/rally/blob/5bf0fed3ddf442d19265d314f8d50bce95ad4127/esrally/log.py#L129).
2. Even if we disable this behavior, the output is slightly verbose:

```
/Users/daniel/Projects/rally/esrally/rally.py:592: FutureWarning: Invoking Rally without a subcommand is deprecated. Specify the 'race' subcommand explicitly.
  warnings.warn("Invoking Rally without a subcommand is deprecated. Specify the 'race' subcommand explicitly.", category=FutureWarning)
```

I've included a note about 2.1.0 in the message in https://github.com/elastic/rally/pull/1155/commits/7513e561a0734a516801d17858b4e400b15e37bf.

Reply:
Thanks for taking a look at the warnings module! This was just something to consider, given the ^^ observations doesn't make sense to use in this project right now.

Also thanks for enhancing the message in https://github.com/elastic/rally/commit/7513e561a0734a516801d17858b4e400b15e37bf, I think it'll help the user base prepare promptly before upgrading to the next minor.

Reply:
Well, thanks for bringing it up! I'll wait until the CI build is finished and will merge the PR then.